src/Analyzer/Passes/SumIfToCountIfPass.cpp

```diff
@@ -54,10 +54,10 @@ class SumIfToCountIfVisitor : public InDepthQueryTreeVisitorWithContext<SumIfToC
             if (!constant_node)
                 return;
 
-            const auto & constant_value_literal = constant_node->getValue();
-            if (!isInt64OrUInt64FieldType(constant_value_literal.getType()))
+            if (auto constant_type = constant_node->getResultType(); !isUInt64(constant_type) && !isInt64(constant_type))
```

Comment:
Now we only added new checks to the `sumIf(...)` branch, but not to the `sum(if(...)). Isn't that a problem? Maybe add tests for both cases, they cannot hurt.